### PR TITLE
[deploy] Update the marketplace csv for use with OLM.

### PR DESCRIPTION
**Pull request by awgreene**
The changes introduced in this PR:
* Rename the marketplace csv so the OLM provided `make release` command includes Marketplace in it's configMap.
* Clearly shows that the latest marketplace image is used
* Removes incorrect x-descriptorsi
* Update the operators display name
* Updates the Readme to reflect these changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ $ export REGISTRY=<SOME_REGISTRY> \
...
+The marketplace operator manages two CRDs: [OperatorSource](./deploy/operatorsource.crd.yaml) and [CatalogSourceConfig](./deploy/catalogsourceconfig.crd.yaml). When an OperatorSource CR is created in the same namespace as where the marketplace operator is running (we recommend the namespace be called "openshift-marketplace"), the operator will download artifacts stored in the registry specified in this OperatorSource CR (for now, please see documentation about using [quay](https://quay.io)'s appregistry API). For an example of this OperatorSource CR please see the [examples](./deploy/examples/) folder.
```
Comment (awgreene) [inline README.md]: Changed "(we recommend the namespace be called "marketplace")" to (we recommend the namespace be called "openshift-marketplace")

File: deploy/marketplace.v0.0.1.clusterserviceversion.yaml [hunk 2]
```diff
@@ -109,11 +109,9 @@ customresourcedefinitions:
...
           path: registryNamespace
       statusDescriptors:
-      - description: The Current phase of the OperatorSource object.
-        displayName: Current Phase
-        path: currentPhase
-        x-descriptors:
-          - 'urn:alm:descriptor:io.kubernetes.phase'
+        - description: The Current phase of the OperatorSource object.
+          displayName: Current Phase
+          path: currentPhase
```
[the remark this answers is not in the document]
Comment (awgreene) [reply deploy/marketplace.v0.0.1.clusterserviceversion.yaml]: Thanks for bringing this up. I reviewed the official OLM documentation for  [building a csv,](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/building-your-csv.md) and it looks like you can omit the `.status.` path.

I then looked at the AMQ Streams operator and saw that only the keafs of their objects had SpecDescriptors, as shown here: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/deploy/chart/catalog_resources/rh-operators/amq-streams.v1.0.0.clusterserviceversion.yaml#L339 (example: description for `kafka.storage.type` but not for `kafka.storage` or `kafka`)

I believe we should update the spec descriptors to describe the leafs of the `CurrentPhase` object, such as the phase name you provided.

[the remark this answers is not in the document]
Comment (awgreene) [reply deploy/marketplace.v0.0.1.clusterserviceversion.yaml]: @aravindhp addressed in latest commit.

File: README.md [hunk 3]
```diff
@@ -54,7 +54,7 @@ $ oc apply -f deploy/crd/operatorsource.crd.yaml
 $ oc apply -f deploy/service_account.yaml
 $ oc apply -f deploy/role.yaml
 $ oc apply -f deploy/role_binding.yaml
-$ oc apply -f deploy/marketplace.csv.yaml
+$ oc apply -f deploy/marketplace.v0.0.1.clusterserviceversion.yaml
 ```
 
 ### Deploying the Marketplace Operator with Kubernetes
```
Comment (kevinrizza) [inline README.md]: There are a couple of references just above this change that use `oc project openshift-marketplace` that need to be updated to `oc project openshift-operators`

Comment (aravindhp) [reply README.md]: Good catch. @awgreene please address that in this commit.
/hold

Comment (awgreene) [reply README.md]: I have addressed this comment in the latest commit.

**Pull request merged**